### PR TITLE
Fix blaze-client handling of HEAD requests

### PR DESCRIPTION
**Pull request by jfwilson**
`blaze-client` was expecting responses to HEAD requests to contain body in the same way as if it had been a GET request, see [this issue](https://github.com/http4s/blaze/issues/44).

### CODE REVIEW

File: blaze-client/src/test/scala/org/http4s/client/blaze/Http1ClientStageSpec.scala [hunk 1]
```diff
@@ -245,6 +245,28 @@ class Http1ClientStageSpec extends Specification {
...
+        LeafBuilder(tail).base(h)
+
+        val response = tail.runRequest(headRequest, false).run
+        response.contentLength must_== Some(contentLength)
+
+        // connection reusable immediately after headers read
+        tail.isRecyclable must_=== true
+
+        // body is empty due to it being HEAD request
+        response.body.runLog.run.foldLeft(0L)(_ + _.length) must_== 0L
```
Comment (rossabaker) [inline blaze-client/src/test/scala/org/http4s/client/blaze/Http1ClientStageSpec.scala]: I'm not going to code golf a test, but I love .runFoldMap:

```scala
body.runFoldMap(_.length).run must_== 0L
```

**Comment by rossabaker** [comment]
Thanks! 👍  I think we can retarget this one to release-0.14.x and cut a bugfix release.

Leaving open for a second set of eyes.

File: blaze-client/src/test/scala/org/http4s/client/blaze/Http1ClientStageSpec.scala (hunk 1 above)
Comment (jfwilson) [inline blaze-client/src/test/scala/org/http4s/client/blaze/Http1ClientStageSpec.scala]: ah nice thanks, didn't know about that method but yes it looks good. I'd need an implicit `Monoid[Long]` in scope, do you know where I can import one of these or do I need to define it myself?
```
implicit val longMonoid = Monoid.instance[Long](_ + _, 0L)
```

**Comment by bryce-anderson** [comment]
This looks good too me as well! 👍
I'll hold off on merging until the runFoldMap business is sorted. If it ends up being more noisy due to having to summon a monoid, I think the current state is fine.

**Comment by rossabaker** [comment]
I'm cleaning up a bunch of chains o' runs in the tests on the fs2 branch anyway.  I'll mop it up there.  The important thing is the bugfix.

**Comment by rossabaker** [comment]
LOL, that changing the base did not do the right thing.  Back to ye olde cherry pick!

File: blaze-client/src/test/scala/org/http4s/client/blaze/Http1ClientStageSpec.scala (hunk 1 above)
Comment (rossabaker) [inline blaze-client/src/test/scala/org/http4s/client/blaze/Http1ClientStageSpec.scala]: For future reference:

* `import Scalaz._`: all batteries included
* `import scalaz.std.anyVal._`: if you prefer minimal imports

**Comment by rossabaker** [comment]
Cherry-picked onto release-0.14.x.  If Travis likes it, I'll cut a release from it.

**Comment by rossabaker** [comment]
release-0.14.x failed because I managed to cherry-pick the second commit onto a detached head.  Let's try this again.

**Comment by rossabaker** [comment]
This is released as v0.14.5 and merged to master.  Thanks again!

**Pull request closed without merging**